src/lib.zig

```diff
@@ -634,7 +634,7 @@ pub fn merkleize(hasher: type, chunks: []chunk, limit: ?usize, out: *[32]u8) any
     // Perform the merkelization
     switch (size) {
         0 => std.mem.copyForwards(u8, out.*[0..], hashes_of_zero[0][0..]),
-        1 => std.mem.copyForwards(u8, out.*[0..], chunks[0][0..]),
+        1 => std.mem.copyForwards(u8, out.*[0..], (if (chunks.len > 0) chunks[0] else hashes_of_zero[0])[0..]),
```

Comment:
i reworded your fix, but this looks wrong
chunks if < next power of two size, are padded with zeros and not hashes of zero.
so empty list of size 1 max list will fail this merkelization

Reply:
ahh my bad, hashes of zero's 0 is zero